dask/dataframe/io/parquet/arrow.py

```diff
@@ -552,33 +598,48 @@ def read_partition(
                 # we slice with `columns` later on.
                 if level not in columns:
                     columns.append(level)
+
+        # Ensure `columns` and `partitions` do not overlap
+        columns_and_parts = columns.copy()
+        if columns_and_parts and partitions:
+            for part_name in partitions.partition_names:
+                if part_name in columns:
+                    columns.remove(part_name)
+                else:
+                    columns_and_parts.append(part_name)
+            columns = columns or None
+
         if isinstance(piece, str):
             # `piece` is a file-path string
-            piece = pq.ParquetDatasetPiece(
-                piece, open_file_func=partial(fs.open, mode="rb")
-            )
+            path = piece
+            row_group = None
+            partition_keys = None
         else:
             # `piece` contains (path, row_group, partition_keys)
             (path, row_group, partition_keys) = piece
+
+        if not isinstance(row_group, list):
+            row_group = [row_group]
+
+        dfs = []
+        for rg in row_group:
             piece = pq.ParquetDatasetPiece(
                 path,
-                row_group=row_group,
+                row_group=rg,
                 partition_keys=partition_keys,
                 open_file_func=partial(fs.open, mode="rb"),
             )
 
-        # Ensure `columns` and `partitions` do not overlap
-        columns_and_parts = columns.copy()
-        if columns_and_parts and partitions:
-            for part_name in partitions.partition_names:
-                if part_name in columns:
-                    columns.remove(part_name)
-                else:
-                    columns_and_parts.append(part_name)
-            columns = columns or None
+            arrow_table = cls._parquet_piece_as_arrow(
+                piece, columns, partitions, **kwargs
+            )
+            df = cls._arrow_table_to_pandas(arrow_table, categories, **kwargs)
+
+            if len(row_group) > 1:
+                dfs.append(df)
 
-        arrow_table = cls._parquet_piece_as_arrow(piece, columns, partitions, **kwargs)
-        df = cls._arrow_table_to_pandas(arrow_table, categories, **kwargs)
+        if len(row_group) > 1:
+            df = pd.concat(dfs)
```

Comment:
What is this for exactly?

Reply:
A single partition may correspond to multiple row groups if the user specifies `split_row_groups>1`.  The ability to aggregate a set number of row-groups into each output DataFrame partition is very important in dask_cudf (where we can read the list of row-groups in a single API call).  Since pyarrow only seems to support a single row-group at a time, we just loop and concat

Reply:
Ah, yes, I see (a small comment might help here).

Pyarrow should in principle support multiple row groups (eg there is a `ParquetFile.read_row_groups`), but this is indeed not exposed in ParquetDatasetPiece. 

When using `pyarrow.dataset`, it is possible to get all row group info's from a Fragment, and then reconstruct a Fragment with any subset (a sinlgle, or multiple) of the row groups.